### PR TITLE
Closes #6221: Allow applications to specify release channel for Socorro

### LINKED ISSUE #6221: Allow applications to specify release channel for Socorro
Currently Socorro is using build config to determine release channel.  However, this sometimes incorrect.  Allow applications to specify release channel for Socorro.

### PR DESCRIPTION
**Pull request by rocketsroger**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/lib/crash/src/test/java/mozilla/components/lib/crash/service/MozillaSocorroServiceTest.kt [hunk 1]
```diff
@@ -469,4 +469,86 @@ class MozillaSocorroServiceTest {
...
+        assert(request.contains("name=Vendor\r\n\r\nMozilla Test"))
+        assert(request.contains("name=Version\r\n\r\n0.0.1"))
+        assert(request.contains("name=ReleaseChannel\r\n\r\ntest channel"))
+
+        verify(service).report(throwable)
+        verify(service).sendReport(throwable, null, null, false, false)
+    }
+
+    @Test
+    fun `MozillaSocorroService reports non specified parameter correctly`() {
```
Comment (Amejia481) [inline components/lib/crash/src/test/java/mozilla/components/lib/crash/service/MozillaSocorroServiceTest.kt]: 🤔  I'm bit confused this test looks exactly the same as `MozillaSocorroService reports specified parameter correctly`? Could you help me to clarify?

Comment (rocketsroger) [reply components/lib/crash/src/test/java/mozilla/components/lib/crash/service/MozillaSocorroServiceTest.kt]: This one is very similar to the other test.  But this is to make sure that we don't accidentally rearrange the parameter order.  I found this issue myself when I accidentally rearranged the parameter order thinking it is OK.

Comment (Amejia481) [reply components/lib/crash/src/test/java/mozilla/components/lib/crash/service/MozillaSocorroServiceTest.kt]: Should the name of the test include something related to the order?

Comment (rocketsroger) [reply components/lib/crash/src/test/java/mozilla/components/lib/crash/service/MozillaSocorroServiceTest.kt]: Agreed.  I'll update the test name.  thanks,

Comment (Amejia481) [reply components/lib/crash/src/test/java/mozilla/components/lib/crash/service/MozillaSocorroServiceTest.kt]: 👍

**Comment by rocketsroger** [comment]
bors r=Amejia481

**Pull request merged**